### PR TITLE
Bring old seed technique back

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

This PR disables the seedable random sampler by default. While yes it can make user results *fully reproducible*, it has a questionable degradation in quality which outweighs its benefits. It can be enabled explicitly in the `Accelerator`. Needs to be paired with a PR in Transformers to configure it there as well. 

TODO: Fix tests

Fixes # (issue)
Fixes https://github.com/huggingface/accelerate/issues/2208

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@BenjaminBossan @SunMarc

### CODE REVIEW

File: src/accelerate/accelerator.py [hunk 1]
```diff
@@ -217,6 +217,9 @@ class Accelerator:
             If set to `True`, in cases where the total batch size across all processes does not exactly divide the
             dataset, samples at the start of the dataset will be duplicated so the batch can be divided equally among
             all workers.
+        use_seedable_sampler (`bool`, *optional*`):
+            Whether or not use a fully seedable random sampler ([`~data_loader.SeedableRandomSampler`]). Comes at a
```
Comment (SunMarc) [inline src/accelerate/accelerator.py:221]: Maybe add a quick note about making the user results fully reproducible

File: src/accelerate/accelerator.py [hunk 2]
```diff
@@ -217,6 +217,9 @@ class Accelerator:
             If set to `True`, in cases where the total batch size across all processes does not exactly divide the
             dataset, samples at the start of the dataset will be duplicated so the batch can be divided equally among
             all workers.
+        use_seedable_sampler (`bool`, *optional*`):
```
Comment (SunMarc) [inline src/accelerate/accelerator.py:220]:
```suggestion
        use_seedable_sampler (`bool`, *optional*, defaults to `False`):
```

**Review by SunMarc — APPROVED** [review]
LGTM ! Just a few nits

**Review by BenjaminBossan — APPROVED** [review]
Makes sense to make this optional, too bad that this seems to cause issues for some users (but we'll probably never hear from users whose training has improved because of the seedable sampler).

Having a test for this would be nice, but could be added later.

**Review by pacman100 — APPROVED** [review]
Thank you @muellerzr for making this configurable!

**Comment by muellerzr** [comment]
Will not merge for now as a few things need to be done (probably fresh in the new year):

1. Run train from scratch examples with known results to verify performance. On my end this is the `imagewoof` and `imagenette` datasets as I know how they behave on untrained models well and they train very quickly
2. Update `Trainer` to have this modification as well in the `TrainingArguments`

**Pull request closed without merging**